### PR TITLE
added type-defs. gitignore-pnpm-lock.yaml. updated rollup-deps

### PR DESCRIPTION
**Pull request by Himujjal**
Still waiting for https://github.com/sveltejs/rfcs/pull/38 to be merged. Once that gets done. I will update the typings again

### CODE REVIEW

**Comment by Himujjal** [comment]
@dummdidumm Hey. Can you merge this PR?

As discussed in https://github.com/sveltejs/svelte-virtual-list/pull/49#issuecomment-869936775

File: package.json [hunk 1]
```diff
@@ -38,5 +36,9 @@
     "src",
     "index.mjs",
     "index.js"
-  ]
+  ],
+  "dependencies": {
+    "@rollup/plugin-commonjs": "^19.0.0",
+    "@rollup/plugin-node-resolve": "^13.0.0"
```
Comment (dummdidumm) [inline package.json:42]: Why did you move these to dependencies? These are needed for bundling and publishing the package, not for users of it.
Since this change seems completely unrelated I suggest to just revert these and only add the `index.d.ts` change.

File: index.d.ts [hunk 2]
```diff
@@ -0,0 +1,13 @@
+declare module "svelte-virtual-list" {
```
Comment (dummdidumm) [inline index.d.ts:1]: The code shouldn't be wrapped inside a `declare module`, like this TS won't pick it up correctly

File: index.d.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  import { SvelteComponentTyped } from "svelte/types/runtime";
```
Comment (dummdidumm) [inline index.d.ts:2]: You can import this from `"svelte"`, no deep import needed

File: index.d.ts [hunk 4]
```diff
@@ -0,0 +1,13 @@
...
+
+  export interface IProps<T> {
+    items: T[];
+    height?: string;
+    itemHeight?: number;
+    start: number;
+    end: number;
+  }
+
+  export default class VirtualList<T> extends SvelteComponentTyped<IProps<T>> {}
```
Comment (dummdidumm) [inline index.d.ts:12]: You should also add a type definition for the slots. That way you can type the generic relationship between `items` and `let:item`

File: index.d.ts [hunk 5]
```diff
@@ -0,0 +1,13 @@
+declare module "svelte-virtual-list" {
+  import { SvelteComponentTyped } from "svelte/types/runtime";
+
+  export interface IProps<T> {
```
Comment (dummdidumm) [inline index.d.ts:4]: Just a personal preference, but I suggest to name the interface `VirtualListProps` instead. Makes it more visible where it's coming from and less prone to name clashes.

**Review by dummdidumm — CHANGES_REQUESTED** [review]
Thanks for taking a stab at this! I added some comments that need to be adressed. Also, the RFC you mention is not needed to establish the generic relationship.

File: package.json (hunk 1 above)
Comment (Himujjal) [reply package.json:42]: ah. didn't notice that. will submit another PR

File: index.d.ts (hunk 3 above)
Comment (Himujjal) [reply index.d.ts:2]: cool.

File: index.d.ts (hunk 5 above)
Comment (Himujjal) [reply index.d.ts:4]: Sure. will do so.
